cmd/ci-operator-configresolver/main.go

```diff
@@ -53,31 +96,48 @@ func validateOptions(o options) error {
 	return nil
 }
 
-func missingQuery(w http.ResponseWriter, field string) {
-	w.WriteHeader(http.StatusBadRequest)
-	fmt.Fprintf(w, "%s query missing or incorrect", field)
+func recordError(label string) {
+	labels := prometheus.Labels{"error": label}
+	configresolverMetrics.errorRate.With(labels).Inc()
+}
+
+func missingQuery(w http.ResponseWriter, field string) (int, []byte) {
+	recordError(fmt.Sprintf("query missing %s", field))
+	return http.StatusBadRequest, []byte(fmt.Sprintf("%s query missing or incorrect", field))
 }
 
 func resolveConfig(agent load.ConfigAgent) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
+		t := time.Now()
+		// get int pointer
+		statusCode := new(int)
+		var body []byte
+		defer func() {
```

Comment:
I think I'd prefer chaining/nesting two `http.HandlerFunc` so that each is simple. Let's look at `traceHandler` in `deck` upstream.